Celeste.Mod.mm/Patches/PlayerPlayback.cs

```diff
@@ -0,0 +1,28 @@
+﻿#pragma warning disable CS0626 // Method, operator, or accessor is marked external and has no attributes on it
+using Celeste.Mod;
+using Microsoft.Xna.Framework;
+using MonoMod;
+using System.Collections.Generic;
+
+namespace Celeste {
+    abstract class patch_PlayerPlayback : PlayerPlayback {
+
+        public patch_PlayerPlayback(EntityData e, Vector2 offset)
+            : base(e, offset) {
+            // no-op. MonoMod ignores this - we only need this to make the compiler shut up.
+        }
+
+        public extern void orig_ctor(EntityData e, Vector2 offset);
+        [MonoModConstructor]
+        public void ctor(EntityData e, Vector2 offset) {
+            string tutorialName = e.Attr("tutorial");
+            if (PlaybackData.Tutorials.ContainsKey(tutorialName)) {
+                orig_ctor(e, offset);
+            } else {
+                patch_LevelEnter.ErrorMessage = Dialog.Get("postcard_missingtutorial").Replace("((tutorial))", tutorialName);
+                Logger.Log(LogLevel.Warn, "PlayerPlayback", $"Failed to load tutorial '{tutorialName}'");
+                throw new KeyNotFoundException();
```

Comment:
This exception should have an error message.

Reply:
I think specific exception messages are useful if we're crashing, but since we'll be catching these I think it's more annoying then anything to just get the same message twice. The generic message at least tells you that it was a dictionary key-access issue which is "new information"

Reply:
Yeah that makes sense.